src/main/resources/assets/js/models/Collection.js

```diff
@@ -0,0 +1,27 @@
+var Backbone = require("backbone");
+
+module.exports = Backbone.Collection.extend({
```

Comment:
I think you can come around this by adding a general function in Marathon.jsx like this:
```
updatePolling: function () {
  var modalClass = this.state.modalClass;
  var id = this.state.activeTabId;
  // logic for checking the state and poll based on that
  },
```
and replace these two with `this.updatePolling()`:
https://github.com/mesosphere/marathon/pull/1066/files#diff-6876cc8f56c80c7276969fd16e24b309R136
https://github.com/mesosphere/marathon/pull/1066/files#diff-6876cc8f56c80c7276969fd16e24b309R146

And move this logic to the `updatePolling` function:
https://github.com/mesosphere/marathon/pull/1066/files#diff-6876cc8f56c80c7276969fd16e24b309L381

That way we can set the state and when the component updates (after setting the state) it will update the polling based on the new state. Does that make sense?